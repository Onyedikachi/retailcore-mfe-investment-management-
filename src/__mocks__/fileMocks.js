module.exports = "";
export let ProviderValue = {
  isChecker: false,
  selected: {
    id: 1,
    text: "created by me",
    value: "created by me",
  },
  setSelected: jest.fn(),
  setIsChecker: jest.fn(),
  category: "all branches",
  setCategory: jest.fn(),
  setStatus: jest.fn(),
  status: "",
  dateData: "",
  setDateData: jest.fn(),
  search: "",
  setSearch: jest.fn(),
  type: "",
  setType: jest.fn(),
  initiator: "",
  setInitiator: jest.fn(),
  setDuration: jest.fn(),
  duration: "",
  isRefreshing: false,
  setRefreshing: jest.fn(),
};
