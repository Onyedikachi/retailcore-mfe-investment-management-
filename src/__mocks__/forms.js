export const countries = [
  {
    id: 1,
    name: "Nigeria",
    value: "Nigeria",
  },
  {
    id: 2,
    name: "Ghana",
    value: "Ghana",
  },
];

export const states = [
  {
    id: 1,
    name: "Lagos",
    value: "Lagos",
  },
  {
    id: 2,
    name: "Ibadan",
    value: "Ibadan",
  },
];

export const LGAs = [
  {
    id: 1,
    name: "Abuja",
    value: "Abuja",
  },
  {
    id: 2,
    name: "Asokoro",
    value: "Asokoro",
  },
];
