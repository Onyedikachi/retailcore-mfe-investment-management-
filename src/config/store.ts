import {
  investmentApi,
  linkApi,
  authApi,
  accountApi,
  currencyApi,
  productMgtApi
} from "../api";
import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import type { PreloadedState } from "@reduxjs/toolkit";

import storage from "redux-persist/lib/storage";
import { combineReducers } from "redux";
import {
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from "redux-persist";
import thunk from "redux-thunk";

const reducers = combineReducers({
  [investmentApi?.reducerPath]: investmentApi?.reducer,
  [linkApi?.reducerPath]: linkApi?.reducer,
  [authApi?.reducerPath]: authApi?.reducer,
  [accountApi?.reducerPath]: accountApi?.reducer,
  [currencyApi?.reducerPath]: currencyApi?.reducer,
  [productMgtApi.reducerPath]: productMgtApi.reducer
});

const persistConfig = {
  key: "root",
  storage,
};

const persistedReducer = persistReducer(persistConfig, reducers);

export const store = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: persistedReducer,
    devTools: process.env.NODE_ENV !== "production",
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        serializableCheck: {
          ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
        },
      })
        .concat(investmentApi?.middleware)
        .concat(linkApi?.middleware)
        .concat(authApi?.middleware)
        .concat(accountApi?.middleware)
        .concat(currencyApi?.middleware)
        .concat(productMgtApi?.middleware),
    preloadedState,
  });
};

export type RootState = ReturnType<typeof persistReducer>;
export type AppStore = ReturnType<typeof store>;
export type AppDispatch = AppStore["dispatch"];
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
