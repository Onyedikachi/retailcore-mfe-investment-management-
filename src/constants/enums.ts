export enum StatusCategoryType {
  AllBranches = "all branches",
  Requests = "requests",
  // Add other category types here
}

export enum StatusItemType {
  Type = "type",
  Request = "request",
}
export enum RequestStatus {
  APPROVED = "Approved",
  INREVIEW = "In-Review",
  INISSUE = "In-Issue",
  DRAFT = "Draft",
}

export enum ProductStatus {
  ACTIVE = "Active",
  INACTIVE = "InActive",
}

export enum Messages {
  ADMIN_BRANCH_SETUP_SENT = "Branch Setup Sent for Approval",
  BRANCH_CREATE_SUCCESS = "Branch created successfully",
  BRANCH_CREATE_FAILED = "Branch creation failed",
  BRANCH_CREATE_APPROVED = "Branch creation approved",
  BRANCH_CREATE_REJECTED = "Branch creation rejected",
  BRANCH_BULK_CREATE_APPROVED = "Branch bulk creation approved",
  BRANCH_BULK_CREATE_REJECTED = "Branch bulk creation rejected",
  ADMIN_BRANCH_CREATE_SUCCESS = "Branch creation request submitted for approval",
  ADMIN_BRANCH_CREATE_FAILED = "Branch creation request failed",
  BRANCH_MODIFY_SUCCESS = "Branch modified successfully",
  BRANCH_MODIFY_FAILED = "Branch modification failed",
  BRANCH_MODIFY_APPROVED = "Branch modification approved",
  BRANCH_MODIFY_REJECTED = "Branch modification rejected",
  ADMIN_BRANCH_MODIFY_SUCCESS = "Branch modification request submitted for approval",
  ADMIN_BRANCH_MODIFY_FAILED = "Branch modification request failed",
  BRANCH_ACTIVATE_SUCCESS = "Branch activated successfully",
  BRANCH_ACTIVATE_FAILED = "Branch activation failed",
  BRANCH_ACTIVATE_APPROVED = "Branch activation approved",
  BRANCH_ACTIVATE_REJECTED = "Branch activation rejected",
  ADMIN_BRANCH_ACTIVATE_SUCCESS = "Branch activation request submitted for approval",
  ADMIN_BRANCH_ACTIVATE_FAILED = "Branch activation request failed",
  BRANCH_DEACTIVATE_SUCCESS = "Branch deactivated successfully",
  BRANCH_DEACTIVATE_FAILED = "Branch deactivation failed",
  BRANCH_DEACTIVATE_APPROVED = "Branch deactivation approved",
  BRANCH_DEACTIVATE_REJECTED = "Branch deactivation rejected",
  ADMIN_BRANCH_DEACTIVATE_SUCCESS = "Branch deactivation request submitted for approval",
  ADMIN_BRANCH_DEACTIVATE_FAILED = "Branch deactivation request failed",
  BRANCH_WITHDRAW_DELETE_SUCCESS = "Request withdrawn and deleted successfully",
  BRANCH_WITHDRAW_DELETE_FAILED = "Unable to withdraw & delete",
  BRANCH_DELETE_SUCCESS = "Request deleted successfully",
  BRANCH_DELETE_FAILED = "Unable to delete",
  BRANCH_APPROVE_FAILED = "Unable to approve branch ",
  BRANCH_REJECT_FAILED = "Unable to reject branch ",
  BRANCH_DRAFT_SUCCESS = "Branch Setup saved to draft",
  BRANCH_DRAFT_FAILED = "Unable to save as draft",
}

export enum Prompts {
  BRANCH_DRAFT = "Do you want to save as draft?",
  BRANCH_DRAFT_SUBTEXT = "Requests in draft will be deleted after 2 hours of inactivity",
  BRANCH_ACTIVATE = "Do you want to activate this branch?",
  BRANCH_DEACTIVATE = "Do you want to deactivate this branch?",
  BRANCH_MODIFY = "Do you want to modify this branch?",
  REMOVE_RECORD = "Do you want to remove this record?",
  CANCEL_PROCESS = "Do you want to cancel process?",
  CANCEL_BRANCH_SETUP = "Do you want to cancel branch setup?",
  BRANCH_CREATION_REJECT = "Do you want to reject branch creation?",
  BRANCH_CREATION_APPROVE = "Do you want to approve branch creation?",
  BRANCH_BULK_CREATION_REJECT = "Do you want to reject bulk branch creation?",
  BRANCH_BULK_CREATION_APPROVE = "Do you want to approve bulk branch creation?",
  BRANCH_MODIFY_REJECT = "Do you want to reject branch modification?",
  BRANCH_MODIFY_APPROVE = "Do you want to approve branch modification?",
  BRANCH_MODIFY_CANCEL = "Do you want to cancel branch modification?",
  BRANCH_WITHDRAW_DELETE = "Do you want to withdraw and delete this request?",
  BRANCH_WITHDRAW_MODIFY = "Do you want to withdraw and modify this request?",
  BRANCH_DELETE = "Do you want to delete this request?",
  BRANCH_DEACTIVATION_APPROVE = "Do you want to approve branch deactivation",
  BRANCH_DEACTIVATION_REJECT = "Do you want to reject branch deactivation",
  BRANCH_ACTIVATION_APPROVE = "Do you want to approve branch activation",
  BRANCH_ACTIVATION_REJECT = "Do you want to reject branch activation",
}
