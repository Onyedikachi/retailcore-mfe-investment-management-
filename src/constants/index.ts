export * from "./permissions";
export * from "./branch";
export * from "./tables";
export * from "./form";
