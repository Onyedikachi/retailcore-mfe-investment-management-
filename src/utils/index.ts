export * from "./capitalizeFirstLetter";
export * from "./capitalizeObjectKeys";
export * from "./context";
export * from "./getPercentage";
export * from "./removeNullFromObj";
export * from "./requestType";
export * from "./ucObjectKeys";
export * from "./handleDropdown";
export * from "./handleColorState";
export * from "./handleUserView";
export * from "./lowerObjectKeys";
