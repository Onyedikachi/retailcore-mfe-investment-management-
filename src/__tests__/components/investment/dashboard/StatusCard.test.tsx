// StatusCard.test.js
import React from "react";
import { render, fireEvent, act } from "@testing-library/react";
import { useEffect } from "react";
import StatusCard, {
  StatusButton,
  sortOptions,
  filterCheckerOptions,
  count,
  handleActiveType,
  handlePermission,
} from "../../../../components/branch-management/dashboard/StatusCard";
import { InvestmentContext } from "../../../../utils/context";
import { StatusCategoryType } from "../../../../constants/enums";
import { StatusTypes, StatusRequests } from "../../../../constants";
import { ProviderValue } from "../../../../__mocks__/fileMocks";
import { Provider } from "react-redux";
import { store } from "../../../../config/store";

const mockSetCategory = jest.fn();
const mockSetSelected = jest.fn();
jest.mock("react-router-dom", () => ({
  BrowserRouter: ({ children }) => <div>{children}</div>,
  Routes: ({ children }) => <div>{children}</div>,
  Route: ({ element }) => element,
  useSearchParams: jest.fn(),
}));
const category = StatusCategoryType?.AllBranches;
jest
  .spyOn(require("react-router-dom"), "useSearchParams")
  .mockReturnValue([new URLSearchParams({ category: "" })]);
function handleSelected(value, setSelected) {
  setSelected(value);
  return value;
}

describe("StatusCard", () => {
  beforeEach(() => {
    mockSetCategory.mockReset();
    mockSetSelected.mockReset();
  });

  it("renders without crashing", () => {
    const setSelected = jest.fn();

    render(
      <Provider store={store()}>
        <InvestmentContext.Provider
          value={{
            ...ProviderValue,
            setSelected,
            setStatus: jest.fn(),
            setCategory: jest.fn,
          }}
        >
          <StatusCard />
        </InvestmentContext.Provider>
      </Provider>
    );
  });

  it("matches snapshot", () => {
    const setSelected = jest.fn();
    const { asFragment } = render(
      <Provider store={store()}>
        <InvestmentContext.Provider
          value={{
            ...ProviderValue,
            setSelected,
            setStatus: jest.fn(),
            setCategory: jest.fn,
          }}
        >
          <StatusCard />
        </InvestmentContext.Provider>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays status categories", () => {
    const setSelected = jest.fn();
    const { getByText } = render(
      <Provider store={store()}>
        <InvestmentContext.Provider
          value={{
            ...ProviderValue,
            setSelected,
            setStatus: jest.fn(),
            setCategory: jest.fn,
          }}
        >
          <StatusCard />
        </InvestmentContext.Provider>
      </Provider>
    );
    expect(getByText("all branches")).toBeInTheDocument();
    expect(getByText("requests")).toBeInTheDocument();
  });

  it("calls setCategory when status category clicked", () => {
    const setSelected = jest.fn();
    const { getByText } = render(
      <Provider store={store()}>
        <InvestmentContext.Provider
          value={{
            ...ProviderValue,
            setSelected,
            setStatus: jest.fn(),
            setCategory: mockSetCategory,
          }}
        >
          <StatusCard />
        </InvestmentContext.Provider>
      </Provider>
    );
    act(() => {
      fireEvent.click(getByText("all branches"));
    });

    expect(mockSetCategory).toHaveBeenCalledWith("all branches");
  });

  it("displays status types for selected category", () => {
    const setSelected = jest.fn();
    const { queryByText, getByText } = render(
      <Provider store={store()}>
        <InvestmentContext.Provider
          value={{
            ...ProviderValue,
            setSelected,
            setStatus: jest.fn(),
            setCategory: jest.fn,
          }}
        >
          <StatusCard />
        </InvestmentContext.Provider>
      </Provider>
    );
    expect(queryByText("Unassigned")).toBeNull();
    fireEvent.click(getByText("requests"));
    expect(queryByText("requests")).toBeInTheDocument();
  });

  it("display all status", () => {
    const setSelected = jest.fn();
    const Render = render(
      <Provider store={store()}>
        <InvestmentContext.Provider
          value={{
            ...ProviderValue,
            setSelected,
            setStatus: jest.fn(),
            setCategory: jest.fn,
          }}
        >
          <StatusCard />
        </InvestmentContext.Provider>
      </Provider>
    );

    expect(Render.getByTestId("all")).toBeInTheDocument();
    expect(Render.getByTestId("approved")).toBeInTheDocument();
    expect(Render.getByTestId("in-issue")).toBeInTheDocument();
    expect(Render.getByTestId("in-review")).toBeInTheDocument();
    expect(Render.getByTestId("draft")).toBeInTheDocument();
  });

  it("calls  setActiveType when button is clicked", () => {
    const setSelected = jest.fn();
    const Render = render(
      <Provider store={store()}>
        <InvestmentContext.Provider
          value={{
            ...ProviderValue,
            setSelected,
            setStatus: jest.fn(),
            setCategory: jest.fn,
          }}
        >
          <StatusCard />
        </InvestmentContext.Provider>
      </Provider>
    );

    fireEvent.click(Render.getByTestId("all"));
    expect(Render.getByTestId("all")).toHaveClass(
      "bg-[#EFEFEF] border-[#D0D5DD]"
    );
  });
});

describe("handleSelected", () => {
  it("should call setSelected with value", () => {
    const setSelected = jest.fn();
    handleSelected("test", setSelected);
    expect(setSelected).toHaveBeenCalledWith("test");
  });

  it("should return the passed value", () => {
    const setSelected = jest.fn();
    const result = handleSelected("test", setSelected);
    expect(result).toBe("test");
  });
});

describe("sortOptions", () => {
  it("returns StatusTypes if category is AllBranches", () => {
    const result = sortOptions(StatusCategoryType.AllBranches, true);
    expect(result).toBe(StatusTypes);
  });

  it("returns StatusRequests if not AllBranches and not isChecker", () => {
    const result = sortOptions(StatusCategoryType.Requests, false);
    expect(result).toBe(StatusRequests);
  });

  // should return an empty array when the 'isChecker' parameter is not a boolean
  it("should return an empty array when isChecker is not a boolean", () => {
    const result = sortOptions(StatusCategoryType.Requests, true);
    expect(result).toHaveLength(4);
  });
});

describe("StatusButton", () => {
  // Tests that the button is rendered with the correct text and count
  it("should render button with correct text and count", () => {
    const item = { type: "all", color: "#000000" };
    const analyticsData = { data: { All: 5 } };
    const { getByText } = render(
      <StatusButton
        item={item}
        isActive={false}
        setActiveType={() => {}}
        analyticsData={analyticsData}
      />
    );
    expect(getByText("all")).toBeInTheDocument();
    expect(getByText("5")).toBeInTheDocument();
  });

  // Tests that clicking the button sets the active type
  it("should set active type when button is clicked", () => {
    const item = { type: "all", color: "#000000" };
    const setActiveType = jest.fn();
    const analyticsData = { data: { All: 5 } };
    const { getByTestId } = render(
      <StatusButton
        item={item}
        isActive={false}
        setActiveType={setActiveType}
        analyticsData={analyticsData}
      />
    );
    fireEvent.click(getByTestId("all"));
    expect(setActiveType).toHaveBeenCalledWith("all");
  });

  // Tests that the button is styled correctly when it is active
  it("should style button correctly when it is active", () => {
    const item = { type: "all", color: "#000000" };
    const analyticsData = { data: { All: 5 } };
    const { getByText } = render(
      <StatusButton
        item={item}
        isActive={true}
        setActiveType={() => {}}
        analyticsData={analyticsData}
      />
    );
    expect(getByText("all")).toHaveClass("font-semibold");
    expect(getByText("all")).toHaveClass("text-sm capitalize font-semibold");
  });

  // Tests that the button is styled correctly when it is inactive
  it("should style button correctly when it is inactive", () => {
    const item = { type: "all", color: "#000000" };
    const analyticsData = { data: { All: 5 } };
    const { getByText } = render(
      <StatusButton
        item={item}
        isActive={false}
        setActiveType={() => {}}
        analyticsData={analyticsData}
      />
    );
    expect(getByText("all")).not.toHaveClass("font-semibold");
    expect(getByText("all")).not.toHaveClass("bg-[#EFEFEF]");
    expect(getByText("all")).toHaveClass("text-sm capitalize font-normal");
  });

  // Tests that the button count is 0 when the count is undefined
  it("should display 0 when count is undefined", () => {
    const item = { type: "all", color: "#000000" };
    const analyticsData = { data: {} };
    const { getByText } = render(
      <StatusButton
        item={item}
        isActive={false}
        setActiveType={() => {}}
        analyticsData={analyticsData}
      />
    );
    expect(getByText("0")).toBeInTheDocument();
  });

  // Tests that the button count is correct when the count is greater than 1
  it("should display correct count when count is greater than 1", () => {
    const item = { type: "all", color: "#000000" };
    const analyticsData = { data: { All: 5 } };
    const { getByText } = render(
      <StatusButton
        item={item}
        isActive={false}
        setActiveType={() => {}}
        analyticsData={analyticsData}
      />
    );
    expect(getByText("5")).toBeInTheDocument();
  });
});

describe("filterCheckerOptions", () => {
  // Tests that the function returns an array with length 4 when called with no arguments.
  it("should return an array with length 4 when called with no arguments", () => {
    const result = filterCheckerOptions();
    expect(result).toHaveLength(4);
  });
});

describe("count", () => {
  // Tests that the 'count' function returns the correct count for the "all" type
  it('should return the correct count for the "all" type', () => {
    const item = { type: "all" };
    const analyticsData = { data: { All: 5 } };
    const result = count(item, analyticsData);
    expect(result).toBe(5);
  });

  // Tests that the 'count' function returns the correct count for the "active" type
  it('should return the correct count for the "active" type', () => {
    const item = { type: "active" };
    const analyticsData = { data: { A: 3 } };
    const result = count(item, analyticsData);
    expect(result).toBe(3);
  });

  // Tests that the 'count' function returns the correct count for the "inactive" type
  it('should return the correct count for the "inactive" type', () => {
    const item = { type: "inactive" };
    const analyticsData = { data: { I: 2 } };
    const result = count(item, analyticsData);
    expect(result).toBe(2);
  });
  // Tests that the 'count' function returns the correct count for the "inactive" type
  it('should return the correct count 0 for the "inactive" type', () => {
    const item = { type: "inactive" };
    const analyticsData = { data: null };
    const result = count(item, analyticsData);
    expect(result).toBe(0);
  });
  it('should return the correct count 0 for the "active" type', () => {
    const item = { type: "active" };
    const analyticsData = { data: null };
    const result = count(item, analyticsData);
    expect(result).toBe(0);
  });

  // Tests that the 'count' function returns the correct count for the "approved" type
  it('should return the correct count for the "approved" type', () => {
    const item = { type: "approved" };
    const analyticsData = { data: { A: 4 } };
    const result = count(item, analyticsData);
    expect(result).toBe(4);
  });

  // Tests that the 'count' function returns the correct count for the "pending" type
  it('should return the correct count for the "pending" type', () => {
    const item = { type: "pending" };
    const analyticsData = { data: { P: 1 } };
    const result = count(item, analyticsData);
    expect(result).toBe(1);
  });

  // Tests that the 'count' function returns the correct count for the "pending" type
  it('should return the correct count for the "rejected" type', () => {
    const item = { type: "rejected" };
    const analyticsData = { data: { R: 1 } };
    const result = count(item, analyticsData);
    expect(result).toBe(1);
  });
  it('should return the correct count for the "rejected" type', () => {
    const item = { type: "rejected" };
    const analyticsData = { data: { R: 1 } };
    const result = count(item, null);
    expect(result).toBe(0);
  });

  // Tests that the 'count' function returns the correct count for the "in-review" type
  it('should return the correct count for the "in-review" type', () => {
    const item = { type: "in-review" };
    const analyticsData = { data: { P: 2 } };
    const result = count(item, analyticsData);
    expect(result).toBe(2);
  });
    // Tests that the 'count' function returns the correct count for the "in-review" type
    it('should return the correct count for the "in-review" type', () => {
      const item = { type: "in-review" };
      const analyticsData = { data: { P: 2 } };
      const result = count(item, null);
      expect(result).toBe(0);
    });
  it("should return the 0 when no tyoe is giving", () => {
    const item = { type: "" };
    const analyticsData = { data: { P: 0 } };
    const result = count(item, analyticsData);
    expect(result).toBe(0);
  });
});

describe("handleActiveType", () => {
  // Tests that the status is set to an empty string when activeType is "all"
  it('should set status to an empty string when activeType is "all"', () => {
    // Arrange
    const activeType = "all";
    let status = "";

    // Act
    handleActiveType(activeType, (value) => {
      status = value;
    });

    // Assert
    expect(status).toBe("");
  });

  // Tests that the status is set to "A" when activeType is "active"
  it('should set status to "A" when activeType is "active"', () => {
    // Arrange
    const activeType = "active";
    let status = "";

    // Act
    handleActiveType(activeType, (value) => {
      status = value;
    });

    // Assert
    expect(status).toBe("A");
  });

  // Tests that the status is set to "I" when activeType is "inactive"
  it('should set status to "I" when activeType is "inactive"', () => {
    // Arrange
    const activeType = "inactive";
    let status = "";

    // Act
    handleActiveType(activeType, (value) => {
      status = value;
    });

    // Assert
    expect(status).toBe("I");
  });

  // Tests that the status is set to "A" when activeType is "approved"
  it('should set status to "A" when activeType is "approved"', () => {
    // Arrange
    const activeType = "approved";
    let status = "";

    // Act
    handleActiveType(activeType, (value) => {
      status = value;
    });

    // Assert
    expect(status).toBe("A");
  });

  // Tests that the status is set to "P" when activeType is "pending"
  it('should set status to "P" when activeType is "pending"', () => {
    // Arrange
    const activeType = "pending";
    let status = "";

    // Act
    handleActiveType(activeType, (value) => {
      status = value;
    });

    // Assert
    expect(status).toBe("P");
  });

  // Tests that the status is set to "D" when activeType is "draft"
  it('should set status to "D" when activeType is "draft"', () => {
    // Arrange
    const activeType = "draft";
    let status = "";

    // Act
    handleActiveType(activeType, (value) => {
      status = value;
    });

    // Assert
    expect(status).toBe("D");
  });
  // Tests that the status is set to "D" when activeType is "draft"
  it('should set status to "R" when activeType is "rejected"', () => {
    // Arrange
    const activeType = "rejected";
    let status = "";

    // Act
    handleActiveType(activeType, (value) => {
      status = value;
    });

    // Assert
    expect(status).toBe("R");
  });
  // Tests that the status is set to "D" when activeType is "draft"
  it('should set status to "P" when activeType is "in-review"', () => {
    // Arrange
    const activeType = "in-review";
    let status = "";

    // Act
    handleActiveType(activeType, (value) => {
      status = value;
    });

    // Assert
    expect(status).toBe("P");
  });

  // Tests that the status is set to "D" when activeType is "draft"
  it('should set status to "R" when activeType is "in-issue"', () => {
    // Arrange
    const activeType = "in-issue";
    let status = "";

    // Act
    handleActiveType(activeType, (value) => {
      status = value;
    });

    // Assert
    expect(status).toBe("R");
  });
});

describe('handlePermission', () => {

  // Sets filtered branch options to BranchOptions if VIEW_ALL_BRANCH_RECORDS permission is present
  it('should set filtered branch options to BranchOptions when VIEW_ALL_BRANCH_RECORDS permission is present', () => {
    const setFilteredRequestOptions = jest.fn();
    const permissions = ['VIEW_ALL_BRANCH_RECORDS'];
    const BranchOptions = [{ value: 'created_by_me' }, { value: 'created_by_anyone' }];
    const RequestOptions = [{ value: 'sent_to_me' }, { value: 'sent_to_anyone' }];
    const setFilteredBranchOptions = jest.fn();

    handlePermission(setFilteredRequestOptions, permissions, BranchOptions, RequestOptions, setFilteredBranchOptions);

    expect(setFilteredBranchOptions).toHaveBeenCalledWith(BranchOptions);
  });

  // Sets filtered request options to RequestOptions if VIEW_ALL_BRANCH_REQUESTS permission is present
  it('should set filtered request options to RequestOptions when VIEW_ALL_BRANCH_REQUESTS permission is present', () => {
    const setFilteredRequestOptions = jest.fn();
    const permissions = ['VIEW_ALL_BRANCH_REQUESTS'];
    const BranchOptions = [{ value: 'created_by_me' }, { value: 'created_by_anyone' }];
    const RequestOptions = [{ value: 'sent_to_me' }, { value: 'sent_to_anyone' }];
    const setFilteredBranchOptions = jest.fn();

    handlePermission(setFilteredRequestOptions, permissions, BranchOptions, RequestOptions, setFilteredBranchOptions);

    expect(setFilteredRequestOptions).toHaveBeenCalledWith(RequestOptions);
  });

  // Does not set filtered branch options if permissions is empty
  it('should not set filtered branch options when permissions is empty', () => {
    const setFilteredRequestOptions = jest.fn();
    const permissions = [];
    const BranchOptions = [{ value: 'created_by_me' }, { value: 'created_by_anyone' }];
    const RequestOptions = [{ value: 'sent_to_me' }, { value: 'sent_to_anyone' }];
    const setFilteredBranchOptions = jest.fn();

    handlePermission(setFilteredRequestOptions, permissions, BranchOptions, RequestOptions, setFilteredBranchOptions);

    expect(setFilteredBranchOptions).not.toHaveBeenCalled();
  });

  // Does not set filtered request options if permissions is empty
  it('should not set filtered request options when permissions is empty', () => {
    const setFilteredRequestOptions = jest.fn();
    const permissions = [];
    const BranchOptions = [{ value: 'created_by_me' }, { value: 'created_by_anyone' }];
    const RequestOptions = [{ value: 'sent_to_me' }, { value: 'sent_to_anyone' }];
    const setFilteredBranchOptions = jest.fn();

    handlePermission(setFilteredRequestOptions, permissions, BranchOptions, RequestOptions, setFilteredBranchOptions);

    expect(setFilteredRequestOptions).not.toHaveBeenCalled();
  });


});
