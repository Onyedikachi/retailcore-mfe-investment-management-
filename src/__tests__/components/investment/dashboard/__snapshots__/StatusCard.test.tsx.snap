// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex border border-[#E5E9EB] rounded-lg"
  >
    <div
      class="bg-[#EFEFEF] w-[208px] rounded-l-lg border-r border-[#D0D5DD] overflow-hidden"
    >
      <button
        class=" hover:bg-gray-50 px-4 py-[19px] text-[18px] text-[#636363] text-left flex gap-x-[5px] items-center leading-[24px] w-full capitalize border-[#D0D5DD] border-b last-of-type:border-b-0"
        data-testid="all branches-btn"
        name="all branches"
        type="button"
      >
        <svg
          class="invisible text-sterling-red-800"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
          />
        </svg>
         all branches
      </button>
      <button
        class=" hover:bg-gray-50 px-4 py-[19px] text-[18px] text-[#636363] text-left flex gap-x-[5px] items-center leading-[24px] w-full capitalize border-[#D0D5DD] border-b last-of-type:border-b-0"
        data-testid="requests-btn"
        name="requests"
        type="button"
      >
        <svg
          class="invisible text-sterling-red-800"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
          />
        </svg>
         requests
      </button>
    </div>
    <div
      class="flex-1 bg-white py-[10px] px-[22px] rounded-r-lg flex justify-between"
    >
      <div
        class="flex gap-x-5 py-[14px] pl-[20px]"
      >
        <button
          class="border min-w-[60px] rounded-lg py-1 px-[15px] flex flex-col cursor-pointer relative bg-[#EFEFEF] border-[#D0D5DD] after:content-[\\"\\"] after:w-1 after:h-[80%] after:absolute after:border-r after:right-[-12px] after:top-1/2 after:translate-y-[-50%] after:border-[#E5E9EB] last-of-type:after:content-none"
          data-testid="all"
        >
          <span
            class="text-sm capitalize font-semibold"
            style="color: rgb(37, 44, 50);"
          >
            all
          </span>
          <span
            class="text-2xl xl:text-4xl text-[#252C32] font-semibold"
          >
            0
          </span>
        </button>
        <button
          class="border min-w-[60px] rounded-lg py-1 px-[15px] flex flex-col cursor-pointer relative border-transparent hover:bg-gray-50 hover:border-[#D0D5DD] after:content-[\\"\\"] after:w-1 after:h-[80%] after:absolute after:border-r after:right-[-12px] after:top-1/2 after:translate-y-[-50%] after:border-[#E5E9EB] last-of-type:after:content-none"
          data-testid="approved"
        >
          <span
            class="text-sm capitalize font-normal"
            style="color: rgb(47, 183, 85);"
          >
            approved
          </span>
          <span
            class="text-2xl xl:text-4xl text-[#252C32] font-semibold"
          >
            0
          </span>
        </button>
        <button
          class="border min-w-[60px] rounded-lg py-1 px-[15px] flex flex-col cursor-pointer relative border-transparent hover:bg-gray-50 hover:border-[#D0D5DD] after:content-[\\"\\"] after:w-1 after:h-[80%] after:absolute after:border-r after:right-[-12px] after:top-1/2 after:translate-y-[-50%] after:border-[#E5E9EB] last-of-type:after:content-none"
          data-testid="in-review"
        >
          <span
            class="text-sm capitalize font-normal"
            style="color: rgb(63, 162, 247);"
          >
            in-review
          </span>
          <span
            class="text-2xl xl:text-4xl text-[#252C32] font-semibold"
          >
            0
          </span>
        </button>
        <button
          class="border min-w-[60px] rounded-lg py-1 px-[15px] flex flex-col cursor-pointer relative border-transparent hover:bg-gray-50 hover:border-[#D0D5DD] after:content-[\\"\\"] after:w-1 after:h-[80%] after:absolute after:border-r after:right-[-12px] after:top-1/2 after:translate-y-[-50%] after:border-[#E5E9EB] last-of-type:after:content-none"
          data-testid="in-issue"
        >
          <span
            class="text-sm capitalize font-normal"
            style="color: rgb(207, 42, 42);"
          >
            in-issue
          </span>
          <span
            class="text-2xl xl:text-4xl text-[#252C32] font-semibold"
          >
            0
          </span>
        </button>
        <button
          class="border min-w-[60px] rounded-lg py-1 px-[15px] flex flex-col cursor-pointer relative border-transparent hover:bg-gray-50 hover:border-[#D0D5DD] after:content-[\\"\\"] after:w-1 after:h-[80%] after:absolute after:border-r after:right-[-12px] after:top-1/2 after:translate-y-[-50%] after:border-[#E5E9EB] last-of-type:after:content-none"
          data-testid="draft"
        >
          <span
            class="text-sm capitalize font-normal"
            style="color: rgb(170, 170, 170);"
          >
            draft
          </span>
          <span
            class="text-2xl xl:text-4xl text-[#252C32] font-semibold"
          >
            0
          </span>
        </button>
      </div>
      <div>
        <div
          class="min-w-[150px]"
          role="combobox"
        >
          <div
            class="relative mt-1"
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              class="relative w-full cursor-pointer rounded-[6px] bg-white py-1 pl-2 pr-10 text-left border border-[#D0D5DD] focus:outline-none  text-[#252C32] text-sm"
              data-headlessui-state=""
              id="headlessui-listbox-button-2"
              type="button"
            >
              <span
                class="block truncate"
              >
                Initiated by me
              </span>
              <span
                class="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-2 text-xs"
              >
                <svg
                  aria-hidden="true"
                  class=" text-[#6E7C87]"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
